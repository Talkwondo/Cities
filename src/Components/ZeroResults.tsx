import React from 'react';
import {Image, Text, View} from 'react-native';
import {zeroResultsStyle} from './style/zeroResultsStyle';
import Logo from '../Assets/logo.png';

export const ZeroResults = () => {
  return (
    <View style={zeroResultsStyle.container} testID="zero-results">
      <Image style={zeroResultsStyle.image} source={Logo} />
      <Text style={zeroResultsStyle.title}>No results</Text>
    </View>
  );
};
