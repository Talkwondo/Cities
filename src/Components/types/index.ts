import {NavigationProp} from '@react-navigation/native';
import {City} from '../../Model/types';
import {Routes} from '../../Model/data/routes';
import {ListRenderItem} from 'react-native';

export interface ButtonCityProps {
  item: City;
  index: number;
  navigation: NavigationProp<RootStackParamList, Routes.CITY>;
  filterLength: number;
}
export type RootStackParamList = {
  MAIN: undefined;
  CITY: City;
};

export interface ItemProps {
  item: ListRenderItem<City>;
  index: number;
}
