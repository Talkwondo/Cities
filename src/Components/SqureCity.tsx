import React from 'react';
import {Text, View} from 'react-native';
import {squareStyle} from './style/squreStyle';
import {City} from '../Model/types';

export const SqureCity = ({city, country, description, background}: City) => {
  return (
    <View style={[squareStyle.container, {backgroundColor: background}]}>
      <Text style={squareStyle.cityHeader}>{city}</Text>
      <Text
        numberOfLines={3}
        ellipsizeMode="tail"
        style={squareStyle.description}>
        {description}
      </Text>
      <Text style={squareStyle.countryHeader}>{country}</Text>
    </View>
  );
};
