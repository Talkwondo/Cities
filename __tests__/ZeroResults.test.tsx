//@ts-ignore
import React = require('react');
import renderer from 'react-test-renderer';
import {ZeroResults} from '../src/Components/ZeroResults';
describe('Snapshot', () => {
  it('to match snapshot', () => {
    const component = renderer.create(<ZeroResults />);
    const match = component.toJSON();
    expect(match).toMatchSnapshot();
  });
});
