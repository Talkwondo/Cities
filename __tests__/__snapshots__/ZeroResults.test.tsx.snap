// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot to match snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="zero-results"
>
  <Image
    source={
      {
        "testUri": "../../../src/Assets/logo.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "height": 60,
        "resizeMode": "center",
        "width": 60,
      }
    }
  />
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#f2f2f2",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
  >
    No results
  </Text>
</View>
`;
